vmdb/app/helpers/ems_cloud_helper/textual_summary.rb

```diff
@@ -46,7 +46,7 @@ def textual_type
   end
 
   def textual_port
-    ["openstack", "openstack_infra", "rhevm"].include?(@ems.emstype) ? {:label => "API Port", :value => @ems.port} : nil
+    @ems.class.supports_port? ? {:label => "API Port", :value => @ems.port} : nil
```

Comment:
I debated that, so yeah I can add that.
On Jan 28, 2015 2:29 PM, "Oleg Barenboim" <notifications@github.com> wrote:

> In vmdb/app/helpers/ems_cloud_helper/textual_summary.rb
> <https://github.com/ManageIQ/manageiq/pull/1476#discussion_r23715193>:
>
> > @@ -46,7 +46,7 @@ def textual_type
> >    end
> >
> >    def textual_port
> > -    ["openstack", "openstack_infra", "rhevm"].include?(@ems.emstype) ? {:label => "API Port", :value => @ems.port} : nil
> > +    @ems.class.supports_port? ? {:label => "API Port", :value => @ems.port} : nil
>
> Should an ems have an instance method supports_port? that delegates to
> the class method? Because, conceptually, there is nothing wrong with asking
> the EMS itself if it supports_port?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ManageIQ/manageiq/pull/1476/files#r23715193>.
>